File: StriverA2Z/Java/Arrays/Medium/LongestConsecutiveSequence.java
import java.util.HashSet;

public class LongestConsecutiveSequence {
    public static void main(String[] args) {
        int[] arr = {100, 1, 2, 101, 102, 2, 3, 4, 4};
        longestSequence(arr);
    }

    static void longestSequence(int[] arr) {
        int longest = 1;
        HashSet<Integer> st = new HashSet<>();
        for (int i : arr) {
            st.add(i);
        }

        for (Integer i : st) {
            if (!st.contains(i - 1)) {
                int cnt = 1;
                int x = i;
                while (st.contains(x + 1)) {
                    cnt++;
                    x++;
                }
                longest = Math.max(longest, cnt);
            }
        }

        System.out.println("The longest consecutive sequence is " + longest);
    }
}

File: StriverA2Z/C++/3_ArraysQuestion/3.2_Medium/longest_consecutive_sequence.cpp
#include<bits/stdc++.h>

using namespace std;

void longest_sequence(vector<int> arr) {
    int longest = 1;
    unordered_set<int> st;
    for (int i: arr) {
        st.insert(i);
    }

    for (int i: st) {
        if (st.find(i - 1) == st.end()) {
            int cnt = 1;
            int x = i;
            while (st.find(x + 1) != st.end()) {
                cnt = cnt + 1;
                x = x + 1;
            }
            longest = max(cnt, longest);
        }
    }

    cout << "The longest consecutive length is " << longest;
}

int main() {
    longest_sequence({100, 4, 200, 1, 3, 2});
    return 0;
}